iceberg/iceberg-common/src/main/java/org/apache/gravitino/iceberg/common/ops/IcebergTableOpsManager.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.iceberg.common.ops;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import java.util.Map;
+import org.apache.commons.lang.StringUtils;
+import org.apache.gravitino.catalog.lakehouse.iceberg.IcebergConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IcebergTableOpsManager implements AutoCloseable {
+  public static final Logger LOG = LoggerFactory.getLogger(IcebergTableOpsManager.class);
+
+  public static final String DEFAULT_CATALOG = "default_catalog";
+  private final Cache<String, IcebergTableOps> icebergTableOpsCache;
+  private final IcebergTableOpsProvider provider;
+
+  public IcebergTableOpsManager(Map<String, String> properties) {
+    this.icebergTableOpsCache = Caffeine.newBuilder().build();
+    this.provider = createProvider(properties);
+    this.provider.initialize(properties);
+  }
+
+  /**
+   * @param rawPrefix The path parameter is passed by a Jetty handler. The pattern is matching
+   *     ([^/]*\/), end with /
+   * @return the instance of IcebergTableOps.
+   */
+  public IcebergTableOps getOps(String rawPrefix) {
+    String prefix = shelling(rawPrefix);
+    String cacheKey = prefix;
+    if (DEFAULT_CATALOG.equals(prefix)) {
+      throw new RuntimeException(
+          String.format("%s is conflict with reserved key, please replace it", prefix));
+    }
+    if (StringUtils.isBlank(prefix)) {
+      LOG.debug("prefix is empty, return default iceberg catalog");
+      cacheKey = DEFAULT_CATALOG;
+    }
```

Comment:
please create a new func `getCatalogName(String prefix)` to get catalog name from prefix. and simple code like this:
```
String catalogName = getCatalogName(String prefix);
return icebergTableOpsCache.get(cacheKey, k -> provider.getIcebergTableOps(catalogName));
```